SNIP-25.md

```diff
@@ -0,0 +1,87 @@
+# SNIP-25 - Improved privacy for [SNIP-20](/SNIP-20.md) tokens
+
+This document outlines updates to the SNIP-20 specification with the introduction of `decoys`, which are designed to strengthen privacy measures by ensuring that account information remains concealed during transaction processes. This is accomplished by intermixing real accounts with 'decoy' accounts, thus making it difficult for potential attackers to discern the actual accounts being updated. Moreover, randomness is integrated into the execution messages, which offers an additional layer of protection by creating uncertainty in storage access patterns.
+
+Contracts that support the existing SNIP-20 standard are still considered
+compliant, and clients SHOULD be written so as to benefit from this feature
+when available.
+
+- [Rationale](#rationale)
+- [Messages](#messages)
+- [Batch Operations](#batch-operations)
+- [Randomness Appendix](#randomness-appendix)
+
+## Rationale
+
+Secret Network smart contracts provide computational privacy through the use of Trusted Execution Environments, which process encrypted data securely without revealing it publicly on chain. However, the patterns at which data is accessed when processing confidential transactions can leak crucial information. In the example of a SNIP-20, each account balance may be stored privately, but the access pattern is public, so an untrusted host OS can potentially learn private information by monitoring accesses to the key-value storage. For example, each user's balance is kept in a key-value store, and the encrypted key for each account stays the same. By keeping track of the keys fetched from the storage, one can figure out the receiver of a SNIP-20 transfer.
```

Comment:
1. "For example, each user's balance is kept in a key-value store, and the encrypted key for each account stays the same." could be misinterpreted to mean that the encryption key stays the same for each account. I recommend clarifying this to something like: "For example, each user's balance is kept in a key-value store, and each account is associated with a consistent, encrypted key."

2. "By keeping track of the keys fetched from the storage, one can figure out the receiver of a SNIP-20 transfer." could be clarified by specifying who or what "one" refers to. A suggested edit would be: "By keeping track of the keys fetched from the storage, an outside observer can figure out the receiver of a SNIP-20 transfer."